File: backend/src/main/java/com/zzang/chongdae/auth/service/dto/SignupResponse.java
package com.zzang.chongdae.auth.service.dto;

import com.zzang.chongdae.member.repository.entity.MemberEntity;

public record SignupResponse(SignupMemberResponseItem member, SignupTokenResponseItem token) {

    public SignupResponse(MemberEntity savedMember, TokenDto tokenDto) {
        this(new SignupMemberResponseItem(savedMember), new SignupTokenResponseItem(tokenDto));
    }
}

File: android/app/src/main/java/com/zzang/chongdae/data/remote/dto/response/SignupResponse.kt
package com.zzang.chongdae.data.remote.dto.response

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SignupResponse(
    @SerialName("member") val memberResponse: MemberResponse,
    @SerialName("token") val tokenResponse: TokenResponse,
)